README.md

```diff
@@ -608,6 +608,54 @@ buildable state (rather than be completely broken).
 
 ## Backend development
 
+### Code Style & Formatting
+
+.NET code style and formatting rules are imposed by [dotnet format](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-format) because it is IDE and platform agnostic, and natively included in the .NET SDK. Rule are dictated by an [.editorconfig file](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/code-style-rule-options).
+
+The warnings and errors about violations of these rules are surfaced at build time by enabling [Enforce Code Style In Build (<EnforceCodeStyleInBuild>)](https://learn.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#enforcecodestyleinbuild) in each `.csproj` file.
+
+To that end - regardless of your IDE of choice - if your solution builds without warnings then your changes should not have caused any style violations. 
+
+*NOTE* We currently have a large number of warnings, many of which we're present before introducing an `.editorconfig` file, and many more of which have been created by imposing style rules we previously weren't. Once the work has been done to resolve these warnings, we can turn on `TreatWarningsAsErrors` to make the build itself fail if there are style violations. 
+
+#### How do I configure my IDE? 
+
+Because the `.editorconfig` file is attached to the solution, both Rider and Visual Studio should automatically use its rules in place of their own defaults.
+
+- on Visual Studio, you may need to visit `Analyze -> Code Cleanup -> Configure Code Cleanup`, then change your default profile to include only the `Fix all warnings and errors set in EditorConfig` step.
+- on Rider, you may need to visit `Settings -> Editor -> Inspection Settings` then ensure `Read settings from editorconfig, project settings and rule sets` is ticked.
+
+#### How do I format only one file?
+
+The easiest way is probably through an IDE, configured as per the step above. Most IDEs can format single files if you right-click it in the explorer.
+Failing that, the faffier harder way is to pass an `--include <PATH>` argument to `dotnet format` on the command line, providing it the files you want it to format.
+
+#### How do I run the formatter manually?
+
+You can perform the same check as the pre-commit hooks / GitHubs actions do and receive a report by running the following command from the repository root:
```

Comment:
We've alluded to pre-commit hooks but doesn't seem like we've added these in yet. It should be possible to do this using the `--include` flag e.g. in the root `package.json`:

```json
{
  "lint-staged":  {
    "*.cs": [
      "dotnet format src/GovUk.Education.ExploreEducationStatistics.sln --include"
    ]
  }
}
```

However, doing this alone doesn't seem to work. Looks like `dotnet format` doesn't work if you try to give absolute paths to `--include`, which is unfortunately what lint-staged will do.

If we change `.husky/pre-commit` to use `--relative` i.e

```
./node_modules/.bin/lint-staged --relative
```

This will now work, but unfortunately this now breaks other pre-commit tasks for JS / TS files i.e. Prettier. Not entirely sure what's going on as absolute paths do work if you manually run the Prettier command with an absolute path.

One solution may be to use a custom script (Node or Bash) that trims the absolute paths into relative ones before passing them to `dotnet format`. For example, using a Node script:

```json
{
  "lint-staged":  {
    "*.cs": [
      "node git-hooks/dotnetFormat.js"
    ]
  }
}
```

Note we have a `git-hooks` directory where stuff like this should live.

Reply:
Quite right, I chose not to commit them in the end. I plan to raise this when we meet for exactly the reasons you mention below about performance. 

As a side note, though, what makes you want to make this part of `lint-staged`? 
I just added the `dotnet format` command to the `pre-commit` file itself and that seemed to work just fine.

Reply:
By having it as part of lint-staged, it allows you to only run the pre-commit task on the specific files that have chnaged. This typically speeds up the pre-commit task significantly as it doesn't need to run across the entire project.

For a really slow formatter like `dotnet format`, I'd say this is a necessity. As mentioned in another comment, a full project format can take over 90 seconds, which I'd consider to be pretty intolerable.

Reply:
Oh, nice! I had absolutely no idea lint-staged only linted the... well, _staged_ files. Well that's rather obvious now I think about it :D 

I always run `pnpm tsc` then `pnpm lint` manually through the console first anyway because I usually do commits through the UI, which means if there are any failures it's a pain getting the console output to see them. Ho hum

Okay well yeah will look into alternatives, then see about adding it to lint staged instead